app/src/androidTest/java/org/mozilla/fenix/ui/robots/SettingsSubMenuSearchRobot.kt

```diff
@@ -181,3 +220,11 @@ private fun addSearchEngineSaveButton() = onView(withId(R.id.add_search_engine))
 private fun assertEngineListContains(searchEngineName: String) {
     onView(withId(R.id.search_engine_group)).check(matches(hasDescendant(withText(searchEngineName))))
 }
+
+private fun threeDotMenu(searchEngineName: String) =
+    onView(
+        allOf(
+            withId(R.id.overflow_menu),
+            withParent(withChild(withText(searchEngineName)))
```

Comment:
Just a comment about this.. If there is not any other way to get this element let's use it but using the hierarchy could cause issues sooner than using other approach in case the UI view changes.

Reply:
In this case there are multiple elements with the same ID (overflow_menu) and the only way I found to differentiate them is by matching the text next to it, which also didn't work with "hasSibling" for some reason I don't know.